apis/oam/v1alpha2/core_types.go

```diff
@@ -0,0 +1,375 @@
+	// paths without a leading dot, for example 'spec.replicas'.
+	FieldPaths []string `json:"fieldPaths"`
+
+	// TODO(negz): Use +kubebuilder:default marker to default Required to false
+	// once we're generating v1 CRDs.
+
+	// Required specifies whether or not a value for this parameter must be
+	// supplied when authoring an ApplicationConfiguration.
+	// +optional
+	Required *bool `json:"required,omitempty"`
+}
+
+// A ComponentSpec defines the desired state of a Component.
+type ComponentSpec struct {
+	// A Workload that will be created for each ApplicationConfiguration that
+	// includes this Component. Workloads must be defined by a
+	// WorkloadDefinition.
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Workload runtime.RawExtension `json:"workload"`
+
+	// Parameters exposed by this component. ApplicationConfigurations that
+	// reference this component may specify values for these parameters, which
+	// will in turn be injected into the embedded workload.
+	// +optional
+	Parameters []ComponentParameter `json:"parameters,omitempty"`
+}
+
+// A ComponentStatus represents the observed state of a Component.
+type ComponentStatus struct {
+	runtimev1alpha1.ConditionedStatus `json:",inline"`
+
+	// TODO(negz): Maintain references to any ApplicationConfigurations that
+	// reference this component? Doing so would allow us to queue a reconcile
+	// for consuming ApplicationConfigurations when this Component changed.
+}
+
+// +kubebuilder:object:root=true
+
+// A Component describes how an OAM workload kind may be instantiated.
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=".spec.workload.kind",name=WORKLOAD-KIND,type=string
+type Component struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ComponentSpec   `json:"spec,omitempty"`
+	Status ComponentStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ComponentList contains a list of Component.
+type ComponentList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Component `json:"items"`
+}
+
+// A ComponentParameterValue specifies a value for a named parameter. The
+// associated component must publish a parameter with this name.
+type ComponentParameterValue struct {
+	// Name of the component parameter to set.
+	Name string `json:"name"`
+
+	// Value to set.
+	Value intstr.IntOrString `json:"value"`
+}
+
+// A ComponentTrait specifies a trait that should be applied to a component.
+type ComponentTrait struct {
+	// A Trait that will be created for the component
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Trait runtime.RawExtension `json:"trait"`
+}
+
+// A ScopeReference refers to an OAM scope resource.
+type ScopeReference struct {
+	// APIVersion of the referenced scope.
+	APIVersion string `json:"apiVersion"`
```

Comment:
I assume that this is the apiversion of a scope CR, not the scopeDefinition, right?

Reply:
Right!